From: Thomas Renninger <trenn@suse.de>
Subject: xen amd numa: Adjust initmem_init to take 4 params
Patch-Mainline: no
References: fate#311968 fate#309727

initmem_init to take 4 params was introduced by this commit from AMD:
patches.arch/0011-x86-Export-k8-physical-topology.patch

This patch got extacted from:
patches.xen/xen3-patch-2.6.33
from the current master branch

Signed-off-by: Thomas Renninger <trenn@suse.de>

---
 arch/x86/kernel/setup-xen.c |   11 +++++++++--
 arch/x86/mm/init_64-xen.c   |    3 ++-
 2 files changed, 11 insertions(+), 3 deletions(-)

Index: linux-2.6.32-SLE11-SP2/arch/x86/kernel/setup-xen.c
===================================================================
--- linux-2.6.32-SLE11-SP2.orig/arch/x86/kernel/setup-xen.c
+++ linux-2.6.32-SLE11-SP2/arch/x86/kernel/setup-xen.c
@@ -795,6 +795,8 @@ static struct dmi_system_id __initdata b
 
 void __init setup_arch(char **cmdline_p)
 {
+	int acpi = 0;
+	int k8 = 0;
 #ifdef CONFIG_XEN
 	unsigned int i;
 	unsigned long p2m_pages;
@@ -1106,10 +1108,15 @@ void __init setup_arch(char **cmdline_p)
 	/*
 	 * Parse SRAT to discover nodes.
 	 */
-	acpi_numa_init();
+	acpi = acpi_numa_init();
 #endif
 
-	initmem_init(0, max_pfn);
+#ifdef CONFIG_K8_NUMA
+	if (!acpi)
+		k8 = !k8_numa_init(0, max_pfn);
+#endif
+
+	initmem_init(0, max_pfn, acpi, k8);
 
 #ifdef CONFIG_ACPI_SLEEP
 	/*
Index: linux-2.6.32-SLE11-SP2/arch/x86/mm/init_64-xen.c
===================================================================
--- linux-2.6.32-SLE11-SP2.orig/arch/x86/mm/init_64-xen.c
+++ linux-2.6.32-SLE11-SP2/arch/x86/mm/init_64-xen.c
@@ -927,7 +927,8 @@ kernel_physical_mapping_init(unsigned lo
 }
 
 #ifndef CONFIG_NUMA
-void __init initmem_init(unsigned long start_pfn, unsigned long end_pfn)
+void __init initmem_init(unsigned long start_pfn, unsigned long end_pfn,
+			 int acpi, int k8)
 {
 	unsigned long bootmap_size, bootmap;
 
